Migrate all client pages to app router

dmijatovic opened the pull request:
# Migrate software, projects and user to app router

**This PR is included in #1641. Please test that PR.**

Related to #1519
Closes #854
Closes #1621
Closes #1622
Closes #1626
Closes #1629

Changes proposed in this pull request:
* Migrate software to app router
* Migrate highlights to app router
* Migrate projects to app router
* Migrate organisation to app router
* Migrate communities to app router
* Migrate persons to app router
* Migrate news to app router
* Migrate user to app router
* Migrate invites to app router. The invite layout of software,project, organisation and community is changed to be in line with rsd invite layout.
* Migrate login pages to app router.
* Migrate admin pages to app router. **Note! I changed order of the admin pages. Based on use I set user related pages at the top (rsd-invites, rsd-users and contributors) as first pages**.
* Includes fix of search using # and & charcters mentioned in previous PR
* Removed _app and _documentation templates from pages folder and some unused functions
* **All routes are migrated from pages to app router.** We only need to migrates api/fe routes and robots.txt 

How to test:
* `make start` to build solution and generate test data
* test software routes: overview, add, edit as rsd-admin, rsd-user and visitor
* test project routes: overview, add, edit as rsd-admin, rsd-user and visitor
* test organisation routes: overview, edit as rsd-admin, rsd-user and visitor
* test communities routes: overview, edit as rsd-admin, rsd-user and visitor
* enable persons module in settings.json. You might need to restart docker to load changes.
* test person routes, overview as rsd-admin and visitor
* test user routes as rsd-admin and rsd-user 
* test all invites for maintainers (software,project,organisation and community) and the rsd invites.
* test login pages: /login, login/failed, login/local
* test working of all admin pages including security (not allowing other users to access admin section).

## Current app router layout approach
There are two layout groups in the app folder: base and container. 
- **base layout group**: does not applies container style to complete page (main element) and has white background (bg-base-100). The components used on the pages that use this layout will apply container layout where needed.
- **container layout group**: applies container style to complete page (main element) and has different background color (bg-base-200)
- software overview page has specific layout
- specific "cascading" layouts within routes: for example "invite" route has additional layout added for all pages of that route. This layout is added to parent layout and applied to all sibling or children pages.
- **I have included AppHeader and AppFooter into main layout**. This required change to homepages where these components are removed. 

**Further improvements of the layout approach might be needed. This is first attempt and all suggestions are welcome.**

PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [ ] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

dmijatovic committed:
feat: migrate organisation pages to app router
chore: migrates organisation overview page
chore: migrated organisation software page
chore: migrated organisation projects page
chore: migrated organisation releases page
chore: migrated organisation units page
chore: migrated organisation settings page

dmijatovic committed:
feat: move communties to app router

dmijatovic committed:
feat: migrate news section to app router
fix: search param encoding/decoding
refactor: extracting search params for software/project filter using shared functions

dmijatovic committed:
feat: migrate persons page to app router

dmijatovic committed:
feat: migrate projects to app router
feat: migrate software to app router
feat: migrate user to app router and refactor
test: some test are fixed and others are disabled/skipped

dmijatovic committed:
fix: encode search param to api to support # and & chars

jmaassen commented:
When I try to download the citation of software in either JATS or Codemeta format, the download fails: 

<img width="865" height="368" alt="Screenshot_2025-10-01_09-07-50" src="https://github.com/user-attachments/assets/d1761408-974e-4f80-8ca1-c25107dc8c3c" />

<img width="858" height="338" alt="Screenshot_2025-10-01_09-06-59" src="https://github.com/user-attachments/assets/6e0833ba-d6bc-4ca3-9cae-bfed88d39097" />

The other formats do work as expected.

jmaassen commented:
When clicking on the "Settings - About" menu item of a community, I sometimes get this: 

<img width="469" height="343" alt="Screenshot_2025-10-01_09-44-03" src="https://github.com/user-attachments/assets/0eff4e4b-a19b-4c97-ba16-448940608575" />

It seems the slug is missing: 

<img width="504" height="143" alt="Screenshot_2025-10-01_09-44-19" src="https://github.com/user-attachments/assets/6d422675-54c0-48da-904a-e2c0478c010f" />

A reload of the page does not help. Moving to other RSD pages, clicking around, and coming back to this one (via My communities) does seem to work: 

<img width="1370" height="772" alt="Screenshot_2025-10-01_09-47-37" src="https://github.com/user-attachments/assets/0b91907a-b6a5-4e61-a37b-5e622b3e0e6d" />

I've seen this happen twice now, but I'm not entirely sure how to reproduce it...

dmijatovic committed:
refactor: apply header and footer to main layout and create not-found page template
fix: highlight filters, filters modal
fix: filter citations array to only items with DOI and version
chore: remove unused components

dmijatovic committed:
e2e: fix failing e2e test by accepting use terms (even for rsd-admin)
chore: fix sonarQube suggestions
chore: bump fe version to v5.3.0 and upgrade dependencies

dmijatovic committed:
feat: migrate invite routes to app router
refactor: move editOrganisation api to organisation folder

dmijatovic committed:
feat: migrate login pages to app router

dmijatovic committed:
feat: migrate custom pages to app router
feat: migrate cookie page to app router
feat: migrate logout page to app router
refactor: add padding to footer for mobiles

dmijatovic committed:
feat: migrate admin section to app folder

dmijatovic committed:
refactor: remove unused files after migration to app router
fix: remove bottom margin from rsd homepage
fix: remove min height on admin custom page when no items
chore: fix sonarcloud issues

ewan-escience reviewed (changes requested):
Everything still seems to work, nice work! Some small remarks:

1. The SonarCloud issues are reasonable and should be fixed.
2. When JavaScript is disabled, the categories on software or project pages have spinners on them while their link is missing.

dmijatovic committed:
fix: do not show loader on categories if javascript is disabled
feat: settings tab on organisation and communities requires JS enabled (skeleton message)
